apps/dapp/src/services/MetricsService.ts

```diff
@@ -83,139 +80,54 @@ export interface SocialMetrics {
-      ENV_VARS.VITE_BACKEND_URL === undefined /*||
-      ENV_VARS.VITE_PUBLIC_FRAX3CRV_F_ADDRESS === undefined ||
-      ENV_VARS.VITE_PUBLIC_FRAX3CRV_F_REWARDS_ADDRESS*/
-    ) {
-      console.info(`
-      VITE_ALCHEMY_PROVIDER_NETWORK=${ENV_VARS.VITE_ALCHEMY_PROVIDER_NETWORK}
-      VITE_ALCHEMY_API_KEY=${ENV_VARS.VITE_ALCHEMY_API_KEY}
-      VITE_PUBLIC_TREASURY_ADDRESS=${ENV_VARS.VITE_PUBLIC_TREASURY_ADDRESS}
-      VITE_SERVER_PRIVATE_KEY=${ENV_VARS.VITE_SERVER_PRIVATE_KEY}
-      VITE_PUBLIC_STABLE_COIN_ADDRESS=${ENV_VARS.VITE_PUBLIC_STABLE_COIN_ADDRESS}
-      VITE_PUBLIC_TEMPLE_ADDRESS=${ENV_VARS.VITE_PUBLIC_TEMPLE_ADDRESS}
-      VITE_PUBLIC_TEMPLE_STAKING_ADDRESS=${ENV_VARS.VITE_PUBLIC_TEMPLE_STAKING_ADDRESS}
-      VITE_PUBLIC_LOCKED_OG_TEMPLE_ADDRESS=${ENV_VARS.VITE_PUBLIC_LOCKED_OG_TEMPLE_ADDRESS}
-      VITE_PUBLIC_TEMPLE_V2_PAIR_ADDRESS=${ENV_VARS.VITE_PUBLIC_TEMPLE_V2_PAIR_ADDRESS}
-      VITE_PUBLIC_TEMPLE_V2_ROUTER_ADDRESS=${ENV_VARS.VITE_PUBLIC_TEMPLE_V2_ROUTER_ADDRESS}
-      VITE_PUBLIC_TEMPLE_AMM_OPS_ADDRESS=${ENV_VARS.VITE_PUBLIC_TEMPLE_AMM_OPS_ADDRESS}
-      VITE_PUBLIC_FRAX3CRV_F_ADDRESS=${ENV_VARS.VITE_PUBLIC_FRAX3CRV_F_ADDRESS}
-      VITE_PUBLIC_FRAX3CRV_F_REWARDS_ADDRESS=${ENV_VARS.VITE_PUBLIC_FRAX3CRV_F_REWARDS_ADDRESS}
-      VITE_BACKEND_URL=${ENV_VARS.VITE_BACKEND_URL}
-      `);
-      //throw new Error(`Missing env vars in Metrics Service`);
-    }
+    this.provider = useProvider();
 
-    const TEMPLE_COIN_ADDRESS = ENV_VARS.VITE_PUBLIC_TEMPLE_ADDRESS;
-    const STABLE_COIN_ADDRESS = ENV_VARS.VITE_PUBLIC_STABLE_COIN_ADDRESS;
-    const ALCHEMY_PROVIDER_NETWORK = ENV_VARS.VITE_ALCHEMY_PROVIDER_NETWORK;
-    const ALCHEMY_API_KEY = ENV_VARS.VITE_ALCHEMY_API_KEY;
-    const TREASURY_ADDRESS = ENV_VARS.VITE_PUBLIC_TREASURY_ADDRESS;
-    const SERVER_PRIVATE_KEY = ENV_VARS.VITE_SERVER_PRIVATE_KEY;
-    const TEMPLE_STAKING_ADDRESS = ENV_VARS.VITE_PUBLIC_TEMPLE_STAKING_ADDRESS;
-    const PAIR_ADDRESS = ENV_VARS.VITE_PUBLIC_TEMPLE_V2_PAIR_ADDRESS;
-    const ROUTER_ADDRESS = ENV_VARS.VITE_PUBLIC_TEMPLE_V2_ROUTER_ADDRESS;
-    const AMM_OPS_ADDRESS = ENV_VARS.VITE_PUBLIC_TEMPLE_AMM_OPS_ADDRESS;
-    const LOCKED_OG_TEMPLE_ADDRESS =
-      ENV_VARS.VITE_PUBLIC_LOCKED_OG_TEMPLE_ADDRESS;
-    const FRAX3CRV_F_ADDRESS = ENV_VARS.VITE_PUBLIC_FRAX3CRV_F_ADDRESS;
-    const FRAX3CRV_F_REWARDS_ADDRESS =
-      ENV_VARS.VITE_PUBLIC_FRAX3CRV_F_REWARDS_ADDRESS;
-    const FARMING_WALLET_ADDRESS = ENV_VARS.VITE_PUBLIC_FARMING_WALLET_ADDRESS;
-    const ENV = ENV_VARS.VITE_ENV;
-
-    this.provider =
-      isDevelopmentEnv()
-        ? new ethers.providers.Web3Provider(window.ethereum)
-        : new ethers.providers.AlchemyProvider(
-            ALCHEMY_PROVIDER_NETWORK,
-            ALCHEMY_API_KEY
-          );
-
-    this.signer = new ethers.Wallet(SERVER_PRIVATE_KEY, this.provider);
-
-    this.stableCoinContract = new ERC20__factory(this.signer).attach(
```

Comment:
removed `signer` altogether and replaced with the static `Factory.connect(address, provider)` method, since everything should be readonly (we also get to ditch the env var SERVER_PRIVATE_KEY from this 👍 )